doc/developer/design/20220303_secrets.md

```diff
@@ -0,0 +1,214 @@
+# Secrets
+
+## Summary
+
+This document fleshes out the design for [**secrets**](../platform/ux.md#secret),
+which securely store sensitive values.
+
+## Design
+
+### Secret SQL syntax and semantics
+
+Four new SQL statements will be introduced with the following grammar to
+create, alter, drop, and list secrets, respectively:
+
+```
+create_secret_stmt ::=
+  CREATE SECRET [IF NOT EXISTS] <name> AS <value>
+
+alter_secret_stmt ::=
+  ALTER SECRET <name> AS <value>
+
+drop_secret_stmt ::=
+  DROP SECRET [IF EXISTS] <name> [{ RESTRICT | CASCADE }]
+
+show_secrets_stmt: SHOW SECRETS [FROM <schema>] [{ LIKE 'pattern' | WHERE <expr> }]
+
+<name> ::= <qualified-identifier>
+<value> ::= <scalar-expression>
+```
+
+**Initially, all of these statements will be available in experimental mode
+only.**
+
+The name of a secret is a qualified identifier. Secrets live in the standard
+database and schema hierarchy alongside tables, views, sources, etc.
+
+The `CREATE SECRET` statement creates a new secret with the specified contents.
+It returns an error if a secret with the specified name already exists unless
+the `IF NOT EXISTS` clause is present. The `<value>` expression may not
+reference any relations and must be implicitly castable to `bytea`.
```

Comment:
what is the best way to send secrets over the wire? This doc says `bytea`. I would have picked a base64 encoded string.